scripts/edge-research/tests/test_data.py

```
@@ -17,3 +17,57 @@ def test_shape_panel_takes_earliest_snapshot_and_computes_ttr():
     assert m1.yes_price == 0.10               # earliest snapshot kept
     assert abs(m1.ttr_days - 10.0) < 1e-6     # 05-29 minus 05-19
     assert set(["yes_price","outcome_yes","market_type","ttr_days","market_score"]).issubset(df.columns)
+
+
+# --- CSV / --datasets-dir mode (offline runner, no DB) ---
+
+def _write_csvs(d):
```

Ruff (0.15.15)

[warning] Missing return type annotation for private function `_write_csvs`

Add return type annotation: `None`

(ANN202)